### PR TITLE
feat(ci): Upload test coverage to codecov

### PR DESCRIPTION
**Pull request by nojnhuh**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -161,6 +161,9 @@ jobs:
           go-version: 1.14
       - name: Test
         run: make go-test-coverage
+      - name: Upload Coverage
+        if: ${{ success() }}
+        run: bash <(curl -s https://codecov.io/bash)
```
Comment (bridgetkromhout) [inline .github/workflows/main.yml:166]: curlbash, fun :)

**Comment by michelleN** [comment]
@nojnhuh what does that line do?

**Comment by nojnhuh** [comment]
This is how they describe it in their docs: https://docs.codecov.io/docs/about-the-codecov-bash-uploader

Looks like there may be a native GitHub Actions solution. I'll look into that.

**Comment by nojnhuh** [comment]
Looks like their GitHub action does basically the same thing. https://github.com/codecov/codecov-action/blob/master/index.js#L127

That looks easier to customize though, so we can start using that once we need it. By itself it doesn't look like we gain anything over the plain script.

**Comment by michelleN** [comment]
ah so we've already exported the CODECOV token?
`export CODECOV_TOKEN="token"`

**Comment by nojnhuh** [comment]
That's only required for private repos.

**Pull request merged**